### ISSUE TITLE
Using SQL Server Compact 4 with .NET Core 3.1 (on Windows only)

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/erikej.github.io#16]

**Comment by nomadstays** [comment]
👍 Thank you.

**Comment by DanielRieck** [comment]
Any chance of working with EF Core too?

**Comment by ErikEJ** [comment]
@DanielRieck That would require someone to sponsor or contribute an update of the EF Core provider to EF Core 3.1 😄

**Comment by ErikEJ** [comment]
@DanielRieck My EF Core 2.2 provider is here: https://github.com/ErikEJ/EntityFramework.SqlServerCompact

**Comment by ashirafumiiro** [comment]
If I wish to use SQL Server Compact in a class library, do I need to also add the assemblies to the client csproj consuming the library?

Thank you!

**Comment by ErikEJ** [comment]
@ashirafumiiro yes, you do.

**Comment by ashirafumiiro** [comment]
Thank you!

**Comment by maxshlain** [comment]
Hi @ErikEj, thanks for your incredible work.
Do you know if by any chance we can make this working in Linux? I'm currently migrating some enterprise dot net core project from Windows to Linux and they would like to continue using SDF files. They have a nice process of extracting application db from SQL server to CE file for incident investigations and they would like to keep that process with minimum frictions
Thanks in advance

**Comment by ErikEJ** [comment]
@maxshlain thanks for the kind words.

SQL Compact is Windows only, it uses a number of Windows only APIs.

**Comment by ThePuterMan** [comment]
Struggling to get Sql Compact 4.0 to run with .NET Core 6.0. Any thoughts?

**Comment by ErikEJ** [comment]
@ThePuterMan I think I need more info than that in order to help! 😄 - like a repro project

**Comment by ThePuterMan** [comment]
PS. I have installed EntityFrameworkCore.SqlServerCompact40

**Comment by ErikEJ** [comment]
@ThePuterMan That wont work! But you can use my EF 6 (Classic) SQLCE provider!

**Comment by ThePuterMan** [comment]
I think I solved it. I will get back to you if not.

**Comment by ErikEJ** [comment]
@ThePuterMan Would like to hear more about your actual issue - a good learning opportunity

**Comment by InteXX** [comment]
@ThePuterMan I also am interesting in knowing more about your scenario and how you solved it.

**Comment by ErikEJ** [comment]
@InteXX What issues are you facing and what are you trying to do?

**Comment by ThePuterMan** [comment]
Hi Erik,

 

As I said before I have fixed the issue by rolling the project back from .NET Core 6.0 to .NET Framework 4.8. The problem was that no matter what I did it refused to recognise SQL Compact.

 

Regards

David

**Comment by ErikEJ** [comment]
I will try to update my sample to .net 6

**Comment by ErikEJ** [comment]
@ThePuterMan @InteXX My sample works fine for me with .NET 6: https://github.com/ErikEJ/SqlCeNetCore/commit/ccbe4107e02bc0d7b36d191420dd59cef3cb1412

**Comment by InteXX** [comment]
@ErikEJ 
> What issues are you facing and what are you trying to do?

I'm trying to run SQLCE under EF Core on .NET 6.

I would love to sponsor you but that's not available to me at the present time.

**Comment by ErikEJ** [comment]
@InteXX I have published an EF6 SQLCE provider  - why not just use that? EF Core 6 SQLCE provider is not going to happen from my side.

**Comment by InteXX** [comment]
@ErikEJ 
> I have published an EF6 SQLCE provider

Yes, and I love it (thank you!). I've been using it for years. But I'm beginning the process of migrating my projects to EF Core.

For SQLCE-based projects, it looks like I'm faced with a tough decision:

1. Stick with EF6 (not TEOTWAWKI)
2. Switch to SQLite (not something to be taken lightly)
3. Write my own provider (last-ditch option, probably won't happen)

**Comment by InteXX** [comment]
@ErikEJ 
> last-ditch option, probably won't happen

That said, if it's not too involved I suppose I might consider giving it a poke (if I had some guidance). Is there much to putting together a provider?

**Comment by ErikEJ** [comment]
@InteXX that sums it up nicely.

Creating a provider is quite involved.

**Comment by InteXX** [comment]
@ErikEJ 
> Creating a provider is quite involved

Aha. I thought so.

**Comment by ThePuterMan** [comment]
Hi Erik, 
 
My apologies I have rolled the code back to .NET framework 4.8 now so I can’t give you any details of the situation at that time. Basically .NET Core 6.0 just did not like SQL Compact. I had other issues as if seems you cannot use My.Settings with .NET Core and the program has settings that need to be saved. Rolling back seemed the most expedient action.
 
Kind regards
David
